File: ps/di-string-match.cpp
class Solution {
public:
    vector<int> diStringMatch(string S) {
        int low = 0, high = S.size(), i;
        vector<int> res;        
        for(i=0;i<S.size();i++) {
            if (S[i] == 'I') res.push_back(low++);
            else res.push_back(high--);
        }
        res.push_back(low);
        return res;
    }
};

File: ps/di-string-match.py
class Solution:
    def diStringMatch(self, s: str) -> List[int]:
        res = []
        i, j = 0, len(s)

        for ch in s:
            if ch == 'I':
                res.append(i)
                i += 1
            else:
                res.append(j)
                j -= 1
        res.append(i)
        return res